CDT_3: ship <tl/expected.hpp> (#9145)

atta-ullah01 opened the pull request:
Replaces the `FetchContent` mechanism with a vendored copy of `tl/expected.hpp` (v1.1.0).

The header is placed in `CGAL/tl/expected.hpp` to avoid namespace pollution.
This also removes the CMake module `CGAL_setup_tl-excepted.cmake`.

## Release Management

* Affected package(s): Constrained_triangulation_3
* Issue(s) solved (if any): #9145
* License and copyright ownership: CC0 1.0 Universal (tl/expected.hpp)

atta-ullah01 committed:
CDT_3: ship <tl/expected.hpp> (#9145)

Replaces the FetchContent mechanism with a vendored copy of tl/expected.hpp (v1.1.0).

The header is placed in CGAL/tl/expected.hpp to avoid namespace pollution.
This also removes the CMake module CGAL_setup_tl-excepted.cmake.

## Release Management

* Affected package(s): Constrained_triangulation_3
* Issue(s) solved (if any): #9145
* License and copyright ownership: CC0 1.0 Universal (tl/expected.hpp)

atta-ullah01 committed:
Installation: move tl/expected.hpp from CDT_3

atta-ullah01 committed:
Add SPDX license tag to expected.hpp

lrineau commented:
> @lrineau and @sloriot : Is there not a problem if someone already uses this header? Should "our" version go into a `CGAL::tl` namespace?

That is true. It could go in the `CGAL::cpp23` namespace. I will make and test the change.

> Is it really needed that we use not yet standard language features?

`std::excepted` was standardized in 2022, and part of C++23. That is a vocabulary type, like `std::any`, or `std::optional`. It is not really new. It can be seen as an improved version of `std::optional`: in case the object is void, an error object is in place, to describe why.

lrineau committed:
update tl/expected

- update to [version 1.3.1](https://github.com/TartanLlama/expected/releases/tag/v1.3.1)
- rename namespace `tl` to `CGAL::cpp23`
- rename the file to `<CGAL/expected.h>`

Once CGAL uses C++23. then that file will use `std::expected`.

lrineau committed:
add copyright

afabri commented:
And what about the `define TL_..."   ?   Will that not trigger already defined errors?

lrineau committed:
fix the detectiong of pkg license

lrineau committed:
rename macros

lrineau committed:
Add Id/URL

lrineau committed:
fix warning about deprecated `egrep`

lrineau committed:
remove trailing whitespaces
